HermaFx.Foundation/RuntimeExtensions/TypeExtensions.cs

```diff
@@ -8,9 +9,40 @@ public static class TypeExtension
 		//a thread-safe way to hold default instances created at run-time
 		private static ConcurrentDictionary<Type, object> typeDefaults = new ConcurrentDictionary<Type, object>();
 
-		public static object GetDefaultValue(this Type type)
+		// Extracted from : http://stackoverflow.com/a/7881481
+		public static object GetDefault(this Type type)
 		{
-			return type.IsValueType ? typeDefaults.GetOrAdd(type, t => Activator.CreateInstance(t)) : null;
+			Guard.IsNotNull(type, nameof(type));
+
+			// If no Type was supplied, if the Type was a reference type, or if the Type was a System.Void, return null
+			if (type == null || !type.IsValueType || type == typeof(void))
```

Comment:
If type == null, just throw.. so does it make any sense to check here for 'type == null'?